document-store/src/main/java/org/hypertrace/core/documentstore/postgres/query/v1/vistors/PostgresFilterTypeExpressionVisitor.java

```diff
@@ -40,14 +42,38 @@ public String visit(final RelationalExpression expression) {
     SelectTypeExpression rhs = expression.getRhs();
 
     // Only an identifier LHS and a constant RHS is supported as of now.
-    PostgresSelectTypeExpressionVisitor lhsParser = new PostgresIdentifierExpressionVisitor();
-    PostgresSelectTypeExpressionVisitor rhsParser = new PostgresConstantExpressionVisitor();
+    PostgresSelectTypeExpressionVisitor lhsVisitor = new PostgresIdentifierExpressionVisitor();
+    PostgresSelectTypeExpressionVisitor rhsVisitor = new PostgresConstantExpressionVisitor();
 
-    String key = lhs.accept(lhsParser);
-    Object value = rhs.accept(rhsParser);
+    String key = lhs.accept(lhsVisitor);
+    Object value = rhs.accept(rhsVisitor);
 
-    return PostgresUtils.parseNonCompositeFilter(
-        key, operator.toString(), value, this.postgresQueryParser.getParamsBuilder());
+    return postgresQueryParser.getPgSelections().containsKey(key)
+        ? PostgresUtils.prepareParsedNonCompositeFilter(
+            postgresQueryParser.getPgSelections().get(key),
+            operator.toString(),
+            value,
+            this.postgresQueryParser.getParamsBuilder())
+        : PostgresUtils.parseNonCompositeFilter(
+            key, operator.toString(), value, this.postgresQueryParser.getParamsBuilder());
```

Comment:
This logic makes it hard to understand without explicit comments.
One way is to add comments explaining why we are doing so.
Another way is to transform the query before parsing (just like we do it for Mongo @ org.hypertrace.core.documentstore.mongo.query.transformer.MongoQueryTransformer).

I prefer the 2nd approach because of 

1. Better debugging (ability to log the original and the transformed query)
2. Can be extended to support all the unsupported querying patterns. [One example is https://github.com/hypertrace/document-store/pull/92, where we support an unsupported operation by transforming the query into a supported form with minimal changes to the parsing logic].
3. Single-class single responsibility. The visitor classes will just be responsible for parsing (translating) the query as it is given, while the transformer classes will be doing all the query transformations.

Reply:
I will do this as the follow-up, first want to identify all the required things for functionality. Added todo.

Reply:
Sure. 👍